src/core/georeferencing.cpp

```diff
@@ -424,7 +424,7 @@ void Georeferencing::save(QXmlStreamWriter& xml) const
 			spec_element.writeAttribute(literal::language, literal::proj_4);
 			xml.writeCharacters(geographic_crs_spec);
 		}
-		if (XMLFileFormat::active_version < 6)
+		if (xml_format_version < 6)
```

Comment:
Well, we may need the active version in various spots while exporting. Introducing the public static `active_version` was just a crude quick fix. Rather than adding a version parameter everywhere, the solution might be making XML stream and version number members of the exporter, and passing the exporter down instead of `QXmlStreamWriter`. At the same time it provides the exporter's `addWarning()`.

Reply:
Passing exporter reference down the chain makes writing unit tests difficult as the test binaries will have to include exporter code in the current scheme. E.g. attempt to write test for `OcdFileImport::importGeoreferencing()` results in unsatisfied link time dependencies as `importGeoreferencing()` lives together with functions handling map objects. This in turn calls for including all source code for map object handling etc. IMHO the way out of this is to either link unit tests against libraries containing the complete code or keep the interfaces lean and standard.

Anyway, discussion about interfaces, although very important in long term, is tangential to the immediate goal of handling OCD georeferencing import. I propose moving it over to openorienteering-devel mailing list.